### ISSUE TITLE
Inheritance with narrower overloads

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#1199]
## Environment data

-   Language Server version: Pylance language server 2020.11.3-pre.1 (pyright 750a296b)
-   OS and version: Mac OS X 11.0.1
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9 (brew)

## Expected behaviour

Should it be possible to specify narrower overloads in derived classes? E.g:

```py
class Base:
    def case(self, value: Any) -> Iterable[Any]:
        return []


class Derived(Base):
    @overload
    def case(self, value: int) -> Iterable[int]:
        ...

    @overload
    def case(self, value: float) -> Iterable[float]:
        ...

    def case(self, value: Any) -> Iterable[Any]:
        return []
```

## Actual behaviour

<img width="1083" alt="Screenshot 2020-11-25 at 18 40 25" src="https://user-images.githubusercontent.com/849479/100263366-b6c0ad80-2f4d-11eb-8dba-43c4c4333bbf.png">

## Logs

Nothing special in the logs.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.89, which I just published. It will also appear in the next Pylance release.
